src/main/java/seedu/address/logic/commands/inventory/InventoryCookCommand.java

```diff
@@ -44,16 +48,39 @@ public InventoryCookCommand(Index targetIndex) {
         this.targetIndex = targetIndex;
     }
 
-    private boolean areIngredientsPresent(Recipe recipe) {
-        // TODO
-        return true;
+    /**
+     * Returns true if all the ingredients in recipe exist in the inventory
+     */
+    private boolean hasInventoryIngredients(ReadOnlyInventory inventory, Recipe recipe) {
+        requireNonNull(inventory);
+        requireNonNull(recipe);
+        ObservableList<Ingredient> inventoryList = inventory.getIngredientList();
+        ObservableList<Ingredient> recipeIngredients = recipe.getIngredients().asUnmodifiableObservableList();
+
+        return recipeIngredients.stream().map(recipeIngredient -> inventoryList
+                .stream().map(inventoryIngredient -> inventoryIngredient.isCompatibleWith(recipeIngredient))
+                .reduce(false, (x, y) -> x || y, (x , y) -> x || y))
+                .allMatch(isCompatible -> isCompatible.equals(true));
     }
 
-    private boolean areIngredientsSufficient(Recipe recipe) {
-        // TODO
-        return true;
+    /**
+     * Returns true if all ingredients's quantity in recipe is lesser than the the same ingredient's quantity
+     * in inventory
+     * The ingredients in other list must exist in this list.
+     */
+    private boolean hasSufficientInventoryIngredients(ReadOnlyInventory inventory, Recipe recipe) {
+        requireNonNull(inventory);
+        requireNonNull(recipe);
+        ObservableList<Ingredient> inventoryList = inventory.getIngredientList();
+        ObservableList<Ingredient> recipeIngredients = recipe.getIngredients().asUnmodifiableObservableList();
+
+        return recipeIngredients.stream().map(recipeIngredient -> inventoryList.stream()
+                .filter(inventoryIngredient -> inventoryIngredient.isCompatibleWith(recipeIngredient))
+                .findFirst().get().sufficientQuantity(recipeIngredient))
+                .allMatch(isSufficient -> isSufficient.equals(true));
```

Comment:
Maybe can use `RecipeInventorySimilarityComparator::calculateSimlarity` here, as the `IngredientQuantity` already has the `asProportionOf` method written for that, then we don't need to add new methods to `IngredientQuantity` and `Ingredient`.

So we can just `return RecipeInventorySimilarityComparator.calculateSimlarity(recipe, inventory) == 1`